---
title: "Clinical traits analysis with TIMBR"
author: "Frederick J. Boehm"
date: "10/24/2019"
output: html_document
---

```{r}
library(tidyverse)
library(TIMBR)
```

```{r read-csv}
(peaks <- readr::read_csv("../data/clinical_suggestive_peak_info.csv"))
traits <- readr::read_csv("../data/pheno_all_clinical_v4_suggestive.csv") %>%
  rename(mouse_id = X1)
```

```{r load-genotypes}
fns <- dir("../data/36-state-genotypes")
geno <- list()
for (i in seq_along(fns)){
  load(file.path("../data/36-state-genotypes", fns[i]))
  geno[[i]] <- prsmth
}
names(geno) <- stringr::str_split_fixed(fns, ".genotype.probs.Rdata", 2)[, 1]
```


```{r prior_M-define}
##### From GK example code
# Specify allelic series prior
# Suggested by Wes
# Influences how much prior weight it places on more or less complicated allelic series
prior_M <- list(model.type = "crp", # crp - Chinese Restaurant Process
                prior.alpha.type = "gamma",
                prior.alpha.shape = 1,
                prior.alpha.rate = 2.333415)
```


```{r load-covariates}
load("../data/reduced_map_qtl2_mapping_objects.Rdata")
```


```{r timbr}
data(mcv.data) # get A matrix
results <- list()
if (!file.exists("../data/timbr-clinical-results.rds")){
  for (peak_num in seq_along(1:nrow(peaks))){
    # get genotypes
    foo <- peaks %>%
      dplyr::select(probe) %>%
      filter(probe == probe[peak_num]) %>%
      purrr::map(.f = function(probe) {lapply(X = geno, FUN = function(gg) gg[rownames(gg) %in% probe, drop = FALSE])}
               ) %>%
      bind_cols() %>%
      as.matrix() %>%
      t() %>%
      as.data.frame() %>%
      as_tibble() %>%
      mutate(mouse_id = names(geno)) %>%
      (function(x){colnames(x)[1:36] <- colnames(geno[[1]]); return(x)}) %>%
      inner_join(traits, by = "mouse_id")
  # covariates
    dat <- addcovar %>%
    as_tibble() %>%
    mutate(mouse_id = rownames(addcovar)) %>%
    inner_join(foo, by = "mouse_id")
  
  # define prior_d
  prior_d <- list(P = dat %>%
                    dplyr::select(AA:GH) %>%
                    as.matrix() %>%
                    (function(x){rownames(x) <- dat$mouse_id; return(x)}),
                  A = mcv.data$prior.D$A, # Describes the mapping from full genoprobs to additive dosages
                  fixed.diplo = FALSE)
  yy <- dat %>%
    dplyr::select(peaks$lodcolumn[peak_num]) %>%
    as.matrix() # Order needs to match genoprobs
  
  ## TIMBR call
  results[[peak_num]] <- TIMBR(y = yy,
                     
                     Z = cbind(1, as.numeric(dat$block2), as.numeric(dat$block3)), 
                   prior.D = prior_d,
                   prior.M = prior_M)
}
  saveRDS(results, "../data/timbr-clinical-results.rds")

} else {
  results <- readRDS("../data/timbr-clinical-results.rds")
}

```




```{r plot-pdf, eval = FALSE}
for (i in seq_along(results)){
  fn <- paste0("../figures/timbr-", peaks$lodcolumn[i], "-Chr", peaks$chr, "-", peaks$Mb, "Mb.pdf")
  pdf(fn)
  TIMBR::TIMBR.plot.haplotypes(results[[i]])
  dev.off()
}
```

```{r plot}
for (i in seq_along(results)){
  TIMBR::TIMBR.plot.haplotypes(results[[i]])
}
```

```{r allele-number}
results %>%
  purrr::map(.f = function(res) hist(res$post.K, plot = TRUE))
```


## Summarizing allele partitioning

We want to partition the eight founder alleles at each QTL. We'll then "collapse" genotypes and redo the scan (over the QTL instead of genome-wide).

```{r tabulate-results}
r1 <- results[[1]]
tibble(partition = names(table(r1$post.M)), count = table(r1$post.M)) %>%
  arrange(desc(count))
```

## Collapse genotype probabilities for first QTL (IL9 on Chr2)

```{r collapse}
gg <- qtl2effects::collapse_geno(genoprobs.1$`2`, c(rep(0, 5), 1, 0, 0))
dimnames(gg)[[2]] <- as.character(0:1)
```

```{r scan1}
trait_name <- "IL9"
trait <- traits %>%
  select(IL9) %>%
  as.matrix() %>%
  (function(x) {rownames(x) <- traits$mouse_id; colnames(x) <- trait_name; return(x)})
s1_out_collapsed <- numeric(length = dim(gg)[3])
s1_out <- numeric(length = dim(gg)[3])

for (i in seq_along(1:dim(gg)[3])){
  s1_out_collapsed[i] <- qtl2::fit1(genoprobs = gg[ , , i], 
            pheno = trait, 
            addcovar = addcovar 
            )$lod
s1_out[i] <- qtl2::fit1(genoprobs = genoprobs.1$`2`[ , , i], 
            pheno = trait, 
            addcovar = addcovar 
            )$lod
  
}
```

## Apply to all elements of `results` list

```{r, scan1_loop}
lods <- list()
for (i in seq_along(1:length(results))){
  r1 <- results[[i]]
  tib <- tibble(partition = names(table(r1$post.M)), 
         count = table(r1$post.M)) %>%
    arrange(desc(count))
  part <- tib$partition[1] %>% # choose the top row in tib
    stringr::str_split(pattern = ",") %>%
    unlist() %>%
    as.numeric()
  gg <- qtl2effects::collapse_geno(genoprobs.1[[peaks$chr[i]]], part)
  dimnames(gg)[[2]] <- as.character(0:max(part))
  ## make trait
  trait_name <- peaks$lodcolumn[i]
  trait <- traits %>%
    select(trait_name) %>%
    as.matrix() %>%
    (function(x) {rownames(x) <- traits$mouse_id; 
    colnames(x) <- trait_name; 
    return(x)}
    )
  
  s1_out_collapsed <- numeric(length = dim(gg)[3])
  s1_out <- numeric(length = dim(gg)[3])
  qtl2::scan1()
  
  
  for (j in seq_along(1:dim(gg)[3])){
    s1_out_collapsed[j] <- qtl2::fit1(genoprobs = gg[ , , j], 
              pheno = trait, 
              addcovar = addcovar 
              )$lod
  s1_out[j] <- qtl2::fit1(genoprobs = genoprobs.1[[peaks$chr[i]]][ , , j], 
              pheno = trait, 
              addcovar = addcovar 
              )$lod
  }
  lods[[i]] <- tibble(collapsed_lod = s1_out_collapsed, lod = s1_out)
}

```

```{r, find_peaks}
sapply(X = lods, FUN = function(x) tibble(max_collapsed_lod = max(x$collapsed_lod),
                                max_lod = max(x$lod)))
```


```{r, perms}
# prep pheno
traits_matrix <- traits %>%
  dplyr::select(- mouse_id) %>%
  as.matrix() %>%
  (function(x){
    rownames(x) <- traits$mouse_id
    return(x)
  })
gg_named <- list() 
gg_named$`18` <- gg
qtl2::scan1perm(gg_named, 
                pheno = traits_matrix, 
                addcovar = addcovar, 
                n_perm = 10000
                )
```



