---
title: "TIMBR: Allele effects in TnSeq Hotspots: One marker per trait for all 10 hotspots"
author: "Frederick J. Boehm"
date: "11/19/2019"
output: 
  github_document: default
  html_document: 
    code_folding: hide
params:
  lastmod: !r lubridate::now()
---

Last modified: `r params$lastmod`.


## Overview

We now consider only one marker per trait-hotspot pair. So, if a Neto trait appears at more than one hotspot, it will be present more than once below. However, if a trait is specific to a single hotspot, I consider it at only one marker, its LOD peak marker within the hotspot.



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r pkgs}
library(dplyr)
library(TIMBR)
```


```{r read-genotypes}
probs <- readRDS("../data/genotypes_array.rds")
traits <- readRDS("../data/tnseq-traits.rds")
```

```{r read-csv}
neto <- read.csv("../data/neto_traits_by_probe3_annotated.csv", stringsAsFactors = FALSE) %>%
  tidyr::pivot_longer(cols = V2:V35, 
                      values_to = "trait", 
                      names_to = "trait_name", 
                      values_drop_na = TRUE, 
                      values_ptypes = list(trait = character())) %>%
  dplyr::select(- trait_name) %>%
  dplyr::mutate(lod = 
           dplyr::select(., probe, trait) %>%
           purrr::pmap_dbl(
           .f = function(probe, trait){
             pheno <- traits[ , colnames(traits) == trait, drop = FALSE]
             geno <- probs[ , , dimnames(probs)[[3]] == probe]
             qtl2::fit1(genoprobs = geno, 
                        pheno = pheno, 
                        )$lod
           }
             )
  ) %>%
    dplyr::select(- cM, - neto.n, - row, - n.traits) %>%
    dplyr::group_by(hs) %>%
    dplyr::group_by(trait, add = TRUE) %>%
    dplyr::filter(lod == max(lod), .preserve = TRUE) %>%
  unique() %>% # needed due to multiple probes having same lod (which coincides with max lod within a group)
  ungroup()
```



## TIMBR setup

```{r prior_M-define}
##### From GK example code
# Specify allelic series prior
# Suggested by Wes
# Influences how much prior weight it places on more or less complicated allelic series
prior_M <- list(model.type = "crp", # crp - Chinese Restaurant Process
                prior.alpha.type = "gamma",
                prior.alpha.shape = 1,
                prior.alpha.rate = 2.333415)
```


```{r load-data-timbr}
data(mcv.data) # get A matrix
```

```{r make_neto_list}
 neto_list <- neto %>%
  dplyr::select(probe, trait) %>%
  apply(FUN = as.list, MARGIN = 1)
```


```{r timbr-call}
outfn <- paste0("../data/timbr-tnseq-neto-traits-one-marker-per-trait-all.rds")
# ensure that inputs to call_timbr all have subjects in same order!
subject_ids <- rownames(traits)
##
indices_gp <- match(subject_ids, rownames(probs))
gp <- probs[indices_gp, , ]
##
if (!file.exists(outfn)){
  timbr_out <- parallel::mclapply(neto_list, 
  #timbr_out <- lapply(neto_small, 
                                  FUN = qtl2effects::call_timbr, 
                                  mc.cores = parallel::detectCores(),
                                  traits_df = traits,
                                  prior_M = prior_M, 
                                  genoprobs_array = gp,
                                  addcovar = NULL
                                  )
  saveRDS(timbr_out, outfn)
} else {
  timbr_out <- readRDS(outfn)
}
names(timbr_out) <- neto$trait
```

```{r annotations}
annots <- neto %>%
  dplyr::mutate(allele_series = purrr::map(timbr_out, .f = function(x){
    pp <- x$p.M.given.y[1]
    return(tibble::tibble(series = names(pp), 
                   probability = pp
                   )
    )
    }
    )
                )
print(annots %>% tidyr::unnest(allele_series), n = 93)
```

The annotations for each plot appear before the plot.

```{r plots}
for (i in seq_along(timbr_out)){
  print(paste0(annots[i, ], collapse = " "))
  TIMBR::TIMBR.plot.haplotypes(timbr_out[[i]])
}
         
         
```

```{r write-csv}
readr::write_csv(annots %>% tidyr::unnest(allele_series), path = "../data/tnseq-neto-traits-annotated-10-hotspots.csv")
```

