---
title: "TnSeq Hotspot 1, 2, 3, 5, 7, 10 (small hotspots)"
author: "Frederick J. Boehm"
date: "11/12/2019"
output: github_document
params:
  lastmod: !r lubridate::now()
---

Last modified: `r params$lastmod`.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r pkgs}
library(tidyverse)
```

```{r read-hot}
fn <- "../data/timbr-tnseq-results-neto-hs1.rds"
timbr_out <- readRDS(fn)
```

```{r hist}
par(mfrow=c(1,2))    # set the plotting area into a 1*2 array
purrr::map(.x = timbr_out, .f = function(x){
  hist(x$post.K)
  TIMBR::TIMBR.plot.haplotypes(x)
}
  )
```


# Pull out the most probable allelic series for each TIMBR analysis

We should add to our annotations table the LOD scores for every trait - marker pair.

```{r}
t1 <- purrr::map(.x = timbr_out, 
           .f = function(x){
             foo <- x$p.M.given.y[1]
             tibble::tibble(posterior_prob = foo, 
                            allele_series = names(foo))
             }
           ) %>%
  dplyr::bind_rows()
```

```{r read-genotypes}
probs <- readRDS("../data/genotypes_array.rds")
```

```{r read-csv}
tt <- read.csv("../data/neto_traits_by_probe3_annotated.csv")
neto <- tt %>%
  tidyr::pivot_longer(cols = V2:V35, values_to = "trait", names_to = "trait_name") %>%
  dplyr::filter(!is.na(trait)) %>%
  dplyr::select(- trait_name, - neto.n, - row, - n.traits, - cM, -hs, - chr)
neto_plus <- tt %>%
  tidyr::pivot_longer(cols = V2:V35, values_to = "trait", names_to = "trait_name") %>%
  dplyr::filter(!is.na(trait)) %>%
  dplyr::select(- trait_name)
```

```{r load-covariates}
load("../data/reduced_map_qtl2_mapping_objects.Rdata")
traits <- readRDS("../data/tnseq-traits.rds")
```

```{r}
neto_plus %>%
  dplyr::filter(hs == 1) %>%
  dplyr::select(probe, trait) %>%
  purrr::pmap( 
           .f = function(probe, trait){
             pheno <- traits[ , colnames(traits) == trait, drop = FALSE]
             geno <- probs[ , , dimnames(probs)[[3]] == probe]
             qtl2::fit1(genoprobs = geno, 
                        pheno = pheno, 
                        addcovar = addcovar
                        )
           }
             ) %>%
  purrr::map(.f = function(x){
    tibble::tibble(lod = x$lod)
  }) %>%
  bind_rows() %>%
  dplyr::bind_cols(neto_plus[1:9, ]) %>%
  dplyr::select(- c(hs, n.traits, lod, neto.n, row), lod) %>%
  dplyr::bind_cols(t1)
```


